presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotFilterExpressionConverter.java

```diff
@@ -371,6 +384,7 @@ public PinotExpression visitSpecialForm(SpecialFormExpression specialForm, Funct
             case SWITCH:
             case WHEN:
             case IS_NULL:
+                return handleIsNull(specialForm, true, context);
```

Comment:
Cases `IF` to `WHEN` will fall in to this block, is that intended?  You would need to move `IS_NULL` below `BIND` to avoid this.

Reply:
Fixed this